config.go

```diff
@@ -1,31 +1,133 @@
+	Relay           string `envconfig:"RELAY" default:"wss://relay.getalby.com/v1"`
+	LNBackendType   string `envconfig:"LN_BACKEND_TYPE"`
+	LNDCertFile     string `envconfig:"LND_CERT_FILE"`
+	LNDMacaroonFile string `envconfig:"LND_MACAROON_FILE"`
+	Workdir         string `envconfig:"WORK_DIR" default:".data"`
+	Port            string `envconfig:"PORT" default:"8080"`
+	DatabaseUri     string `envconfig:"DATABASE_URI" default:"nostr-wallet-connect.db"`
+	CookieSecret    string `envconfig:"COOKIE_SECRET"`
+}
+
+type UserConfig struct {
+	ID        uint   `gorm:"primaryKey"`
+	Key       string `gorm:"unique;uniqueIndex;not null"`
+	Value     string
+	Encrypted bool
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
 type Config struct {
-	// These config can be set either by .env or the database config table.
-	// database config always takes preference.
-	db.Config
-	CookieSecret            string `envconfig:"COOKIE_SECRET"`
-	CookieDomain            string `envconfig:"COOKIE_DOMAIN"`
-	ClientPubkey            string `envconfig:"CLIENT_NOSTR_PUBKEY"`
-	Relay                   string `envconfig:"RELAY" default:"wss://relay.getalby.com/v1"`
-	PublicRelay             string `envconfig:"PUBLIC_RELAY"`
-	LNDCertFile             string `envconfig:"LND_CERT_FILE"`
-	LNDMacaroonFile         string `envconfig:"LND_MACAROON_FILE"`
-	BreezWorkdir            string `envconfig:"BREEZ_WORK_DIR" default:".breez"`
-	BasicAuthUser           string `envconfig:"BASIC_AUTH_USER"`
-	BasicAuthPassword       string `envconfig:"BASIC_AUTH_PASSWORD"`
-	Port                    string `envconfig:"PORT" default:"8080"`
-	DatabaseUri             string `envconfig:"DATABASE_URI" default:"nostr-wallet-connect.db"`
-	DatabaseMaxConns        int    `envconfig:"DATABASE_MAX_CONNS" default:"10"`
-	DatabaseMaxIdleConns    int    `envconfig:"DATABASE_MAX_IDLE_CONNS" default:"5"`
-	DatabaseConnMaxLifetime int    `envconfig:"DATABASE_CONN_MAX_LIFETIME" default:"1800"` // 30 minutes
-	IdentityPubkey          string
+	Env            *AppConfig
+	CookieSecret   string
+	NostrSecretKey string
+	NostrPublicKey string
+	db             *gorm.DB
+}
+
+func (cfg *Config) Init(db *gorm.DB, env *AppConfig) {
+	cfg.db = db
+	cfg.Env = env
+
+	if cfg.Env.Relay != "" {
+		cfg.SetUpdate("Relay", cfg.Env.Relay, "")
+	}
+	if cfg.Env.LNBackendType != "" {
+		cfg.SetUpdate("LNBackendType", cfg.Env.LNBackendType, "")
+	}
+	if cfg.Env.LNDCertFile != "" {
+		certBytes, err := os.ReadFile(cfg.Env.LNDCertFile)
+		if err != nil {
+			certHex := hex.EncodeToString(certBytes)
+			cfg.SetUpdate("LNDCertHex", certHex, "")
+		}
+	}
+	if cfg.Env.LNDMacaroonFile != "" {
+		macBytes, err := os.ReadFile(cfg.Env.LNDMacaroonFile)
+		if err != nil {
+			macHex := hex.EncodeToString(macBytes)
+			cfg.SetUpdate("LNDMacaroonHex", macHex, "")
+		}
+	}
+	// set the cookie secret to the one from the env
+	// if no cookie secret is configured we create a random one and store it in the DB
+	cfg.CookieSecret = cfg.Env.CookieSecret
```

Comment:
Should this be encrypted too?

Reply:
I don't think so. those are also not easy to encrypt with a user password because it's needed before the user could enter somthing